lib/mat_SDCZ.ml

```diff
@@ -158,6 +158,60 @@ let to_col_vecs mat =
     for i = 2 to n do ar.(i - 1) <- col mat i done;
     ar
 
+let of_col_vecs_list = function
+  | [] -> create 0 0
+  | (h :: _) as lst ->
+    let n = List.length lst in
+    let m = Array1.dim h in
+    let mat = create m n in
+    let rec loop i = function
+      | [] -> mat
+      | h :: t ->
+        if Array1.dim h <> m then
+          failwith "of_col_vecs_list: vectors not of same length";
+        direct_copy ~n:m ~ofsy:1 ~incy:1 ~y:(col mat i) ~ofsx:1 ~incx:1 ~x:h;
+        loop (i + 1) t
+    in
+    loop 1 lst
+
+let to_col_vecs_list mat =
+  let n = dim2 mat in
+  let rec loop i acc =
+    if i < 1 then acc
+    else
+      loop (i - 1) (col mat i :: acc)
+  in
+  loop n []
+
+let of_list = function
+  | [] -> create 0 0
+  | (h :: _) as lst ->
+    let m = List.length lst in
+    let n = List.length h in
+    let mat = create m n in
+    let rec loop i = function
+      | [] -> mat
+      | h :: t ->
+        List.iteri (fun jm1 e -> mat.{i, jm1+1} <- e) h;
+        loop (i + 1) t
+    in
+    try loop 1 lst
+    with Invalid_argument _ ->
+      failwith "of_list: vectors not of same length"
+
+let to_list mat =
+  let m = dim1 mat in
+  let n = dim2 mat in
+  let row_to_list r =
```

Comment:
I've changed `to_list` to avoid unnecessary closure allocations.  I'm not too concerned if a function performs a constant number of small allocations for convenience and readability.  The OCaml runtime is pretty darn good at handling that.  But the submitted `to_list` allocates the closure passed to `List.iteri` for every single row.  If you have a "tall and slim" matrix (e.g. millions of rows and only two columns), this would lead to a measurable slowdown and put extra pressure on the GC.